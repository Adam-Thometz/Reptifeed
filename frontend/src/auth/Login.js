import React, { useState } from "react";
import { useNavigate } from 'react-router-dom';
import Alert from "../common/Alert";
import './Login.css'

const Login = ({ login }) => {
  const navigate = useNavigate();
  document.title = "Log in | Reptifeed";

  const init = {
    username: '',
    password: ''
  }; 

  const [formData, setFormData] = useState(init);
  const [formMessages, setFormMessages] = useState([]);

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(d => ({ ...d, [name]: value }));
  };

  const handleSubmit = async e => {
    e.preventDefault();
    const res = await login(formData);
    if (res.success) {
      navigate(`/users/${res.id}`);
    } else {
      setFormMessages(res.errors);
    };
  };

  return (
    <div tabIndex={0} className="Login">
      <h1>Login to your Reptifeed account</h1>
      <form onSubmit={handleSubmit}>
        <div className="Login-form-group">
          <label className="Login-label" htmlFor="username">Username:</label>
          <input
            className="Login-input"
            type='text'
            id='username'
            name='username'
            value={formData.username}
            onChange={handleChange}
          />
        </div>
        <div className="Login-form-group">
          <label className="Login-label" htmlFor="password">Password:</label>
          <input
            className="Login-input"
            type='password'
            id='password'
            name='password'
            value={formData.password}
            onChange={handleChange}
          />
        </div>
        <button className="Login-submit" type="submit">Login</button>
        <p>Want to try this app?</p>
        <p>Username: abc</p>
        <p>Password: 123</p>
        {formMessages.length ? <Alert type="danger" messages={formMessages} /> : null }
      </form>
    </div>
  )
};

export default Login;