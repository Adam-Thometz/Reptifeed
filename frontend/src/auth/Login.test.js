import React from "react";
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Login from "./Login";

test('it renders without crashing', () => {
  render(<MemoryRouter>
    <Login />
  </MemoryRouter>);
});

test('it matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Login />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});