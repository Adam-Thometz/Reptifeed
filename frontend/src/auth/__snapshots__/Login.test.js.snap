// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <h1>
      Login to your Reptifeed account
    </h1>
    <form>
      <div
        class="Login-form-group"
      >
        <label
          class="Login-label"
          for="username"
        >
          Username:
        </label>
        <input
          class="Login-input"
          id="username"
          name="username"
          type="text"
          value=""
        />
      </div>
      <div
        class="Login-form-group"
      >
        <label
          class="Login-label"
          for="password"
        >
          Password:
        </label>
        <input
          class="Login-input"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <button
        class="Login-submit"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
